docs/painless/painless-basic-syntax.asciidoc

```diff
@@ -0,0 +1,304 @@
+[[painless-basic-syntax]]
+=== Basic Syntax
+
+[[comments]]
+==== Comments
+
+Painless supports both single-line and multi-line comments. Comments can be
+included anywhere within a script. Use the `//` token anywhere on a line to
+specify a single-line comment. All characters from the `//` token to the end
+of the line are ignored. Use an opening `/*` token and a closing `*/` token
+to specify a multi-line comment. Multi-line comments can start anywhere on a
+line, and all characters in between the `/*` token and `*/` token are ignored.
+
+*Grammar:*
+[source,ANTLR4]
+----
+SINGLE_LINE_COMMENT: '//' .*? [\n\r];
```

Comment:
If you are really into including bits of the grammar you can actually include it at documentation build time with a construct like this one:
```
["source","ANTLR4",subs="attributes,callouts,macros"]
--------------------------------------------------
include-tagged::{parser}[comments]
--------------------------------------------------
```

It'd be a neat thing to do in a followup because it'd make sure the grammar never goes out of date.

Reply:
I will keep this in mind.  Some of the grammar has been modified for the documentation and does not directly follow the current grammar out of convenience for the reader.